sno/working_copy/base.py

```diff
@@ -149,32 +153,36 @@ def table_identifier(self, dataset_or_table):
         )
         return self.preparer.format_table(sqlalchemy_table)
 
+    def _table_def_for_column_schema(self, col, dataset):
+        # This is just used for selects/inserts/updates - we don't need the full type information.
+        # We only need type information if some automatic conversion needs to happen on read or write.
+        # TODO: return the full type information so we can also use it for CREATE TABLE.
+        return sqlalchemy.column(col.name)
+
     @functools.lru_cache()
-    def table_def_for_dataset(self, dataset):
+    def _table_def_for_dataset(self, dataset, schema=None):
```

Comment:
Happily they are both actually immutable -
- a Schema is immutable because I made it immutable
- a Dataset is immutable because you made it immutable! and because it is kind of a view of a particular git tree, which is immutable. Like a git tree, it can't be edited directly - instead you just have to use information from it to help build a new commit.